File: BPMOPMobile_VietNamAirLines/BPMOPMobile_VietNamAirLines/BPMOPMobile.Droid/Class/CustomFlexBoxRecyclerView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Support.V7.Widget;
using Android.Util;
using Android.Views;
using Android.Widget;

namespace BPMOPMobile.Droid.Class
{
    /// <summary>
    /// Custom Recyclerview này giúp cho không vượt quá 3 Line của Choose User
    /// </summary>
    public class CustomFlexBoxRecyclerView : RecyclerView
    {
        int _rowHeight = -1, _maxRow = -1;
        public void SetMaxRowAndRowHeight(int _rowHeight, int _maxRow)
        {
            this._rowHeight = _rowHeight;
            this._maxRow = _maxRow;
        }
        public CustomFlexBoxRecyclerView(Context context, IAttributeSet attrs) : base(context, attrs)
        {
        }

        public CustomFlexBoxRecyclerView(Context context, IAttributeSet attrs, int defStyle) : base(context, attrs, defStyle)
        {
        }

        protected CustomFlexBoxRecyclerView(IntPtr javaReference, JniHandleOwnership transfer) : base(javaReference, transfer)
        {
        }

        protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
        {
            if (_rowHeight != -1 && _maxRow != -1)
            {
                heightMeasureSpec = MeasureSpec.MakeMeasureSpec(_rowHeight * _maxRow, MeasureSpecMode.AtMost);
            }
            else
            {
                heightMeasureSpec = MeasureSpec.MakeMeasureSpec(95 * 3, MeasureSpecMode.AtMost);
            }
            base.OnMeasure(widthMeasureSpec, heightMeasureSpec);
        }

    }
}

File: VNABPMOP/app/src/main/java/com/vuthao/bpmop/base/custom/CustomFlexBoxRecyclerView.java
package com.vuthao.bpmop.base.custom;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

public class CustomFlexBoxRecyclerView extends RecyclerView {
    int _rowHeight = -1, _maxRow = -1;

    public CustomFlexBoxRecyclerView(@NonNull Context context) {
        super(context);
    }

    public CustomFlexBoxRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomFlexBoxRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setMaxRowAndRowHeight(int _rowHeight, int _maxRow) {
        this._rowHeight = _rowHeight;
        this._maxRow = _maxRow;
    }

    @Override
    protected void onMeasure(int widthSpec, int heightSpec) {
        if (_rowHeight != -1 && _maxRow != -1) {
            heightSpec = MeasureSpec.makeMeasureSpec(_rowHeight * _maxRow, MeasureSpec.AT_MOST);
        } else {
            heightSpec = MeasureSpec.makeMeasureSpec(95 * 3, MeasureSpec.AT_MOST);
        }
        super.onMeasure(widthSpec, heightSpec);
    }
}
